lib/marked-renderer.js

```diff
@@ -1,4 +1,15 @@
 const { marked } = require('marked')
+const { gfmHeadingId } = require('marked-gfm-heading-id')
+
+// mangle will default to false in future versions of marked.
+// Explicitly set it to avoid deprecation warnings
+marked.use({
```

Comment:
Unfortunately, we can't just update the custom renderer's constructor with these options, as the renderer itself doesn't do anything with `mangle` or plugins, so we have to target the parent marked instance.

Reply:
Do you think this could be a problem?

By changing the `marked` package we also override the options/plugins for these two:

https://github.com/alphagov/govuk-design-system/blob/91d85fa09cc41c8fc99bfd9663cad46f349635c0/lib/extract-page-headings/index.js#L11

https://github.com/alphagov/govuk-design-system/blob/91d85fa09cc41c8fc99bfd9663cad46f349635c0/lib/get-macro-options/index.js#L8

Looks like the constructor works but this line causes the majority of the deprecation warnings:

https://github.com/alphagov/govuk-design-system/blob/91d85fa09cc41c8fc99bfd9663cad46f349635c0/lib/get-macro-options/index.js#L34

Could either add the options there too:

```js
option.description = marked(option.description, {
  headerIds: false,
  mangle: false,
  renderer
})
```

Or update this other `renderer` to use `new DesignSystemRenderer()` as well:

```js
option.description = renderer.html(option.description)
```

Reply:
Right, I've tweaked things to address this comment!